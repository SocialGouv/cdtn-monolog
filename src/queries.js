import * as es from "./elastic";
import * as Suggestion from "./analysis/suggestion";
import * as Covisit from "./analysis/covisit";

const Queries = (esClient, index) => {
  return {
    getSuggestions: async () => {
      const suggestionsQuery = {
        term: { reportType: Suggestion.reportType },
      };
      return es.getDocuments(esClient, index, suggestionsQuery);
    },

    getCovisitLinks: async (content) => {
      const covisitQuery = {
        bool: {
          must: [
            {
              term: {
                reportType: Covisit.reportType,
              },
            },
            {
              term: {
                content,
              },
            },
          ],
        },
      };

      //   console.log(JSON.stringify(covisitQuery, null, 2));

      const res = await es.getDocuments(esClient, index, covisitQuery);
      if (res.length == 1) {
        return res[0];
      } else {
        throw new Error(`Covisit query failed for content : ${content}`);
      }
    },
  };
};

export { Queries };
