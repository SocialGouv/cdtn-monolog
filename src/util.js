export const SERVICE_URL = "https://code.travail.gouv.fr/";

export const urlToPath = (url) => {
  if (url.startsWith(SERVICE_URL)) {
    return url.slice(SERVICE_URL.length);
  } else {
    return url;
  }
};

export const actionTypes = {
  home: "home",
  search: "search",
  // FIXME: 28/07 should be changed to select_related in the near future
  // selectRelated: "selectRelated",
  selectRelated: "select_related",
  selectResult: "select_result",
  selectSuggestion: "select_suggestion",
  themes: "themes",
  visit: "visit_content",
};
