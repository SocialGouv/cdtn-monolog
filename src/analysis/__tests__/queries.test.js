import { DataFrame } from "data-forge";

// import * as util from "../../util";
// import * as path from "path";
import { actionTypes } from "../../util";
import { analyse, buildCache } from "../queries";

describe("Query reports", () => {
  const toSearchSelectionAction = (uvi, query) => ({
    query,
    type: actionTypes.search,
    uvi,
  });

  const generate = (visit) =>
    new DataFrame(visit.map(([uvi, ss]) => toSearchSelectionAction(uvi, ss)));

  it("should generate cache for each search query", async () => {
    const visits = [
      "paternite",
      "conges maternite",
      "conge de maternite",
    ].map((query, i) => [[i, query]]);

    const df = DataFrame.concat(visits.map(generate));

    const cache = await buildCache(df);

    expect(cache).toMatchSnapshot();
  });

  it("should keep counts for searches and selections in every visit", async () => {
    const visits = [
      "paternite",
      "conges maternite",
      "conge de maternite",
    ].map((query, i) => [[i, query]]);

    const df = DataFrame.concat(visits.map(generate));

    const reports = await analyse(df);
    expect(reports).toMatchSnapshot();
  });

  it("should compute scores for each query cluster", () => {});

  it("should generate the report from data", () => {});
});
