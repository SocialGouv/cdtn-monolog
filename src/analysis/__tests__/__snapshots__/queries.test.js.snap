// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query reports should generate cache for each search query 1`] = `
Object {
  "queryGroups": Array [
    Array [
      "paternite",
    ],
    Array [
      "conges maternite",
    ],
    Array [
      "conge de maternite",
    ],
  ],
  "resultCache": Map {
    "paternite" => Array [
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "3f4a1c5605",
        "description": "Lors de la naissance d'un enfant ou de son hospitalisation immédiate, la personne salariée vivant avec la mère bénéficie d'un congé indemnisé.",
        "slug": "conge-de-paternite-et-daccueil-de-lenfant-dun-salarie-du-secteur-prive",
        "source": "fiches_service_public",
        "title": "Congé de paternité et d'accueil de l'enfant d'un salarié du secteur privé",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F3156",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "3bb8c6e46c",
        "description": "Le père salarié ainsi que, le cas échéant, le conjoint salarié de la mère ou la personne salariée liée à elle par un PACS ou vivant maritalement avec elle bénéficient d’un congé de paternité et d’accu",
        "slug": "le-conge-de-paternite-et-daccueil-de-lenfant#Quelle-est-la-duree-du-conge-de-paternite-et-d-accueil-de-l-enfant-nbsp",
        "source": "fiches_ministere_travail",
        "title": "Le congé de paternité et d’accueil de l’enfant : quelle est la durée du congé de paternité et d’accueil de l’enfant ?",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "ac18f4c3a4",
        "description": "Le salarié qui souhaite bénéficier du congé de paternité et d’accueil de l’enfant avertit son employeur au moins un mois avant la date à laquelle il envisage de le prendre, en précisant la date à laqu",
        "slug": "le-conge-de-paternite-et-daccueil-de-lenfant#Quelles-sont-les-formalites",
        "source": "fiches_ministere_travail",
        "title": "Le congé de paternité et d’accueil de l’enfant : quelles sont les formalités ?",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "9d6cf790da",
        "description": "Pendant le congé de paternité, le contrat de travail est suspenduLe salaire n’est pas maintenu. Toutefois, s’il cesse toute activité, le bénéficiaire du congé peut percevoir des indemnités journalière",
        "slug": "le-conge-de-paternite-et-daccueil-de-lenfant#Quels-effets-sur-le-contrat-de-travail",
        "source": "fiches_ministere_travail",
        "title": "Le congé de paternité et d’accueil de l’enfant : quels effets sur le contrat de travail ?",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "c8e4f480b1",
        "description": "Les demandeurs d’emploi et les stagiaires de la formation professionnelle peuvent bénéficier d’un congé de paternité et d’accueil de l’enfant.Demandeurs d’emploiLes personnes bénéficiant d’un congé de",
        "slug": "le-conge-de-paternite-et-daccueil-de-lenfant#Demandeurs-d-emploi-stagiaires-de-la-formation-professionnelle-des-nbsp",
        "source": "fiches_ministere_travail",
        "title": "Le congé de paternité et d’accueil de l’enfant : demandeurs d’emploi, stagiaires de la formation professionnelle… des conditions particulières ?",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "76ef24f331",
        "description": "Le congé de naissance et le congé de paternité et d'accueil peuvent se succéder si le salarié respecte les conditions lui permettant de les enchaîner.",
        "slug": "peut-on-enchainer-conge-de-naissance-et-conge-de-paternite-et-daccueil",
        "source": "fiches_service_public",
        "title": "Peut-on enchaîner congé de naissance et congé de paternité et d'accueil ?",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F12647",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Départ de l'entreprise",
            "slug": "/themes/8-depart-de-lentreprise",
          },
          Object {
            "label": "Licenciement",
            "slug": "/themes/84-licenciement",
          },
          Object {
            "label": "Licenciement pour motif personnel",
            "slug": "/themes/842-licenciement-pour-motif-personnel",
          },
        ],
        "cdtnId": "83133cf1d2",
        "description": "Le jeune père salarié bénéficie d'une protection contre le licenciement durant les 10 semaines qui suivent la naissance de son enfant.",
        "slug": "le-jeune-pere-salarie-beneficie-t-il-dune-protection-contre-le-licenciement",
        "source": "fiches_service_public",
        "title": "Le jeune père salarié bénéficie-t-il d'une protection contre le licenciement ?",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F32292",
      },
    ],
    "conges maternite" => Array [
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "79448100ac",
        "description": "L'employeur applique les conditions prévues par la convention ou l'accord collectif ou un usage, si elles sont plus favorables que le code du travail pour…",
        "slug": "quelle-est-la-duree-du-conge-de-maternite",
        "source": "contributions",
        "title": "Quelle est la durée du congé de maternité ?",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "c7c50b7ee8",
        "description": "Pour être indemnisée, vous devez remplir les conditions suivantes :

être affiliée à la Sécurité sociale depuis au moins 10 mois à la date présumée de…",
        "slug": "quelles-sont-les-conditions-dindemnisation-pendant-le-conge-de-maternite",
        "source": "contributions",
        "title": "Quelles sont les conditions d’indemnisation pendant le congé de maternité ? ",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "8328342034",
        "description": "La salariée enceinte bénéficie d'un congé maternité durant la période qui se situe autour de la date présumée de son accouchement.",
        "slug": "conge-maternite-dune-salariee-du-secteur-prive",
        "source": "fiches_service_public",
        "title": "Congé maternité d'une salariée du secteur privé",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F2265",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "59b5e2591a",
        "description": "À l’issue de son congé de maternité, la salariée retrouve son précédent emploi ou un emploi similaire assorti d’une rémunération au moins équivalente (sur les garanties offertes à la salariée en matiè",
        "slug": "le-conge-de-maternite#Quelle-est-la-situation-a-l-issue-du-conge",
        "source": "fiches_ministere_travail",
        "title": "Le congé de maternité : quelle est la situation à l’issue du congé ?",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "74f35e3ab2",
        "description": "La salariée peut, quelle que soit son ancienneté dans l’entreprise, suspendre son contrat de travail pendant une durée fixée comme indiqué dans le tableau suivant (il convient également de se reporter",
        "slug": "le-conge-de-maternite#Quelle-est-la-duree-du-conge-de-maternite",
        "source": "fiches_ministere_travail",
        "title": "Le congé de maternité : quelle est la durée du congé de maternité ?",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "adea392c00",
        "description": "Plusieurs dispositions spécifiques sont prévues pour faire face à des situations particulières.Ainsi : en cas d’hospitalisation de l’enfant d’une durée supérieure à 6 semaines après sa naissance, la m",
        "slug": "le-conge-de-maternite#Quelles-sont-les-dispositions-specifiques",
        "source": "fiches_ministere_travail",
        "title": "Le congé de maternité : quelles sont les dispositions spécifiques ?",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Santé, sécurité et conditions de travail",
            "slug": "/themes/6-sante-securite-et-conditions-de-travail",
          },
          Object {
            "label": "Travailleurs et secteurs spécifiques",
            "slug": "/themes/64-travailleurs-et-secteurs-specifiques",
          },
          Object {
            "label": "Femme enceinte - Maternité",
            "slug": "/themes/642-femme-enceinte-maternite",
          },
        ],
        "cdtnId": "14d889edfd",
        "description": "La loi prévoit uniquement l'obligation d'informer l'employeur avant de partir en congé maternité.",
        "slug": "une-salariee-enceinte-est-elle-obligee-de-reveler-sa-grossesse-a-son-employeur",
        "source": "fiches_service_public",
        "title": "Une salariée enceinte est-elle obligée de révéler sa grossesse à son employeur ?",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F1144",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Départ de l'entreprise",
            "slug": "/themes/8-depart-de-lentreprise",
          },
          Object {
            "label": "Licenciement",
            "slug": "/themes/84-licenciement",
          },
        ],
        "cdtnId": "2014cd215a",
        "description": "Le licenciement est interdit pendant la période de congé de maternité. Cependant, avant et après le congé maternité, le licenciement est autorisé en cas de faute grave ou d'impossibilité de maintenir le contrat pour un motif étranger à la grossesse et à l'accouchement.",
        "slug": "licenciement-dune-salariee-enceinte-ou-en-conge-de-maternite",
        "source": "fiches_service_public",
        "title": "Licenciement d'une salariée enceinte ou en congé de maternité",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F2873",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Départ de l'entreprise",
            "slug": "/themes/8-depart-de-lentreprise",
          },
          Object {
            "label": "Démission",
            "slug": "/themes/81-demission",
          },
        ],
        "cdtnId": "184ebe1ef6",
        "description": "Oui. Toute salariée peut démissionner pendant un congé maternité, dans les conditions habituelles prévues en matière de démission.",
        "slug": "peut-on-demissionner-pendant-un-conge-maternite",
        "source": "fiches_service_public",
        "title": "Peut-on démissionner pendant un congé maternité ?",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F1730",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "f25f92dafb",
        "description": "Le congé de maternité peut être fractionné en cas d'hospitalisation prolongée de l'enfant au-delà de la 6ème semaine suivant sa naissance.",
        "slug": "hospitalisation-du-nouveau-ne-quelles-consequences-sur-le-conge-de-maternite",
        "source": "fiches_service_public",
        "title": "Hospitalisation du nouveau-né : quelles conséquences sur le congé de maternité ?",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F32967",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "b16c71fcb0",
        "description": "Le salarié peut bénéficier de congés parentaux successifs selon certaines conditions.",
        "slug": "peut-on-beneficier-de-plusieurs-conges-parentaux-successifs",
        "source": "fiches_service_public",
        "title": "Peut-on bénéficier de plusieurs congés parentaux successifs ?",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F12809",
      },
    ],
    "conge de maternite" => Array [
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "79448100ac",
        "description": "L'employeur applique les conditions prévues par la convention ou l'accord collectif ou un usage, si elles sont plus favorables que le code du travail pour…",
        "slug": "quelle-est-la-duree-du-conge-de-maternite",
        "source": "contributions",
        "title": "Quelle est la durée du congé de maternité ?",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "c7c50b7ee8",
        "description": "Pour être indemnisée, vous devez remplir les conditions suivantes :

être affiliée à la Sécurité sociale depuis au moins 10 mois à la date présumée de…",
        "slug": "quelles-sont-les-conditions-dindemnisation-pendant-le-conge-de-maternite",
        "source": "contributions",
        "title": "Quelles sont les conditions d’indemnisation pendant le congé de maternité ? ",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "8328342034",
        "description": "La salariée enceinte bénéficie d'un congé maternité durant la période qui se situe autour de la date présumée de son accouchement.",
        "slug": "conge-maternite-dune-salariee-du-secteur-prive",
        "source": "fiches_service_public",
        "title": "Congé maternité d'une salariée du secteur privé",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F2265",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "59b5e2591a",
        "description": "À l’issue de son congé de maternité, la salariée retrouve son précédent emploi ou un emploi similaire assorti d’une rémunération au moins équivalente (sur les garanties offertes à la salariée en matiè",
        "slug": "le-conge-de-maternite#Quelle-est-la-situation-a-l-issue-du-conge",
        "source": "fiches_ministere_travail",
        "title": "Le congé de maternité : quelle est la situation à l’issue du congé ?",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "74f35e3ab2",
        "description": "La salariée peut, quelle que soit son ancienneté dans l’entreprise, suspendre son contrat de travail pendant une durée fixée comme indiqué dans le tableau suivant (il convient également de se reporter",
        "slug": "le-conge-de-maternite#Quelle-est-la-duree-du-conge-de-maternite",
        "source": "fiches_ministere_travail",
        "title": "Le congé de maternité : quelle est la durée du congé de maternité ?",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "adea392c00",
        "description": "Plusieurs dispositions spécifiques sont prévues pour faire face à des situations particulières.Ainsi : en cas d’hospitalisation de l’enfant d’une durée supérieure à 6 semaines après sa naissance, la m",
        "slug": "le-conge-de-maternite#Quelles-sont-les-dispositions-specifiques",
        "source": "fiches_ministere_travail",
        "title": "Le congé de maternité : quelles sont les dispositions spécifiques ?",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Santé, sécurité et conditions de travail",
            "slug": "/themes/6-sante-securite-et-conditions-de-travail",
          },
          Object {
            "label": "Travailleurs et secteurs spécifiques",
            "slug": "/themes/64-travailleurs-et-secteurs-specifiques",
          },
          Object {
            "label": "Femme enceinte - Maternité",
            "slug": "/themes/642-femme-enceinte-maternite",
          },
        ],
        "cdtnId": "14d889edfd",
        "description": "La loi prévoit uniquement l'obligation d'informer l'employeur avant de partir en congé maternité.",
        "slug": "une-salariee-enceinte-est-elle-obligee-de-reveler-sa-grossesse-a-son-employeur",
        "source": "fiches_service_public",
        "title": "Une salariée enceinte est-elle obligée de révéler sa grossesse à son employeur ?",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F1144",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Départ de l'entreprise",
            "slug": "/themes/8-depart-de-lentreprise",
          },
          Object {
            "label": "Licenciement",
            "slug": "/themes/84-licenciement",
          },
        ],
        "cdtnId": "2014cd215a",
        "description": "Le licenciement est interdit pendant la période de congé de maternité. Cependant, avant et après le congé maternité, le licenciement est autorisé en cas de faute grave ou d'impossibilité de maintenir le contrat pour un motif étranger à la grossesse et à l'accouchement.",
        "slug": "licenciement-dune-salariee-enceinte-ou-en-conge-de-maternite",
        "source": "fiches_service_public",
        "title": "Licenciement d'une salariée enceinte ou en congé de maternité",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F2873",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Départ de l'entreprise",
            "slug": "/themes/8-depart-de-lentreprise",
          },
          Object {
            "label": "Démission",
            "slug": "/themes/81-demission",
          },
        ],
        "cdtnId": "184ebe1ef6",
        "description": "Oui. Toute salariée peut démissionner pendant un congé maternité, dans les conditions habituelles prévues en matière de démission.",
        "slug": "peut-on-demissionner-pendant-un-conge-maternite",
        "source": "fiches_service_public",
        "title": "Peut-on démissionner pendant un congé maternité ?",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F1730",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "f25f92dafb",
        "description": "Le congé de maternité peut être fractionné en cas d'hospitalisation prolongée de l'enfant au-delà de la 6ème semaine suivant sa naissance.",
        "slug": "hospitalisation-du-nouveau-ne-quelles-consequences-sur-le-conge-de-maternite",
        "source": "fiches_service_public",
        "title": "Hospitalisation du nouveau-né : quelles conséquences sur le congé de maternité ?",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F32967",
      },
      Object {
        "_score": 0,
        "algo": "pre-qualified",
        "breadcrumbs": Array [
          Object {
            "label": "Congés et repos",
            "slug": "/themes/4-conges-et-repos",
          },
          Object {
            "label": "Congés",
            "slug": "/themes/41-conges",
          },
          Object {
            "label": "Congés liés à la naissance et à l'enfance",
            "slug": "/themes/412-conges-lies-a-la-naissance-et-a-lenfance",
          },
        ],
        "cdtnId": "b16c71fcb0",
        "description": "Le salarié peut bénéficier de congés parentaux successifs selon certaines conditions.",
        "slug": "peut-on-beneficier-de-plusieurs-conges-parentaux-successifs",
        "source": "fiches_service_public",
        "title": "Peut-on bénéficier de plusieurs congés parentaux successifs ?",
        "url": "https://www.service-public.fr/particuliers/vosdroits/F12809",
      },
    ],
  },
}
`;
